frontend/src/util/normalizeArray.ts

```diff
@@ -0,0 +1,18 @@
+import _ from "lodash";
+
+export function normalizeArray(
+    arr: number[],
+    new_min: number,
+    new_max: number,
+): number[] {
+    if (arr.length == 0) {
+        return [];
+    }
+    const min = _.min(arr) as number;
+    const max = _.max(arr) as number;
+
+    return arr.map(
+        (value) =>
+            ((value - min) / (max - min)) * (new_max - new_min) + new_min,
+    );
```

Comment:
this is 3 potential total passes of the array, however, I think its possible to rearrange this equation to allow this to only be two passes. as you do the first pass of the array you change each value and also store values to compare for min and max. then you do a second pass now knowing the max and min. however this is a waste of time so this is fine :)

Reply:
you are completely correct and i thank you for taking the time to write this out but i will not be implementing it because im lazy :) (also yeah it does sorta feel like a waste of time)